core/src/Kokkos_View.hpp

```diff
@@ -788,618 +788,295 @@ class View : public ViewTraits<DataType, Properties...> {
-
-#define KOKKOS_IMPL_VIEW_OPERATOR_VERIFY(ARG)                             \
-  Kokkos::Impl::runtime_check_memory_access_violation<                    \
-      typename traits::memory_space>(                                     \
-      "Kokkos::View ERROR: attempt to access inaccessible memory space"); \
-  Kokkos::Impl::view_verify_operator_bounds<typename traits::memory_space> ARG;
+#define KOKKOS_IMPL_VIEW_OPERATOR_VERIFY(...)                               \
+  Kokkos::Impl::runtime_check_memory_access_violation<                      \
+      typename traits::memory_space>(                                       \
+      "Kokkos::View ERROR: attempt to access inaccessible memory space");   \
+  Kokkos::Impl::view_verify_operator_bounds<typename traits::memory_space>( \
+      __VA_ARGS__);
 
 #else
 
-#define KOKKOS_IMPL_SINK(ARG)
-
-#define KOKKOS_IMPL_VIEW_OPERATOR_VERIFY(ARG)          \
+#define KOKKOS_IMPL_VIEW_OPERATOR_VERIFY(...)          \
   Kokkos::Impl::runtime_check_memory_access_violation< \
       typename traits::memory_space>(                  \
       "Kokkos::View ERROR: attempt to access inaccessible memory space");
 
 #endif
 
- public:
-  //------------------------------
-  // Rank 0 operator()
-
-  template <int R            = Rank,
-            class /*Enable*/ = std::enable_if_t<(0 == Rank) && (R == Rank)>>
-  KOKKOS_FORCEINLINE_FUNCTION reference_type operator()() const {
-    KOKKOS_IMPL_VIEW_OPERATOR_VERIFY((m_track, m_map))
-    return m_map.reference();
+  template <typename... Is>
+  static KOKKOS_FUNCTION void check_access_member_function_valid_args(Is...) {
+    static_assert(Rank <= sizeof...(Is), "");
+    static_assert(sizeof...(Is) <= 8, "");
+    static_assert(Kokkos::Impl::are_integral<Is...>::value, "");
   }
-  //------------------------------
-  // Rank 1 operator()
 
-  template <typename I0>
-  KOKKOS_FORCEINLINE_FUNCTION
-      typename std::enable_if<(Kokkos::Impl::are_integral<I0>::value &&
-                               (1 == Rank) && !is_default_map),
-                              reference_type>::type
-      operator()(const I0& i0) const {
-    KOKKOS_IMPL_VIEW_OPERATOR_VERIFY((m_track, m_map, i0))
-    return m_map.reference(i0);
+  template <typename... Is>
+  static KOKKOS_FUNCTION void check_operator_parens_valid_args(Is...) {
+    static_assert(Rank == sizeof...(Is), "");
+    static_assert(Kokkos::Impl::are_integral<Is...>::value, "");
   }
 
+ public:
+  //------------------------------
+  // Rank 1 default map operator()
+
   template <typename I0>
   KOKKOS_FORCEINLINE_FUNCTION
-      typename std::enable_if<(Kokkos::Impl::are_integral<I0>::value &&
-                               (1 == Rank) && is_default_map &&
-                               !is_layout_stride),
-                              reference_type>::type
-      operator()(const I0& i0) const {
-    KOKKOS_IMPL_VIEW_OPERATOR_VERIFY((m_track, m_map, i0))
+      std::enable_if_t<(Kokkos::Impl::always_true<I0>::value &&  //
```

Comment:
Why the trailing empty comment (I see that in a number of places)?

Reply:
Cheap way to influence clang-format.  Do you want me to remove them?

Reply:
No; they are fine.